AEPServices/Tests/services/HttpConnectionTests.swift

```diff
@@ -0,0 +1,127 @@
+        let connection = HttpConnection(data: nil, response: response, error: nil)
+        
+        // test
+        let iOS13Result = connection.responseHttpHeader(forKey: searchKey)
+        let iOS12Result = connection.response?.lowercasedHeaders[searchKey.lowercased()] as? String
+        
+        // verify
+        XCTAssertEqual("value", iOS13Result)
+        XCTAssertEqual(iOS13Result, iOS12Result)
+    }
+    
+    func testResponseHttpHeaderDuplicateKeys() throws {
+        // setup
+        let searchKey = "some-key"
+        let response = HTTPURLResponse(url: URL(string: "https://someurl")!, statusCode: 200, httpVersion: nil, headerFields: [
+            "Some-Key": "value",
+            "some-key": "anotherValue"])
+        let connection = HttpConnection(data: nil, response: response, error: nil)
+        
+        // test
+        let iOS13Result = connection.responseHttpHeader(forKey: searchKey)
+        let iOS12Result = connection.response?.lowercasedHeaders[searchKey.lowercased()] as? String
+        
+        // verify
+        /// in alignment with os behavior for `value(forHTTPHeaderField:)`,
+        /// requesting a header with duplicate entries is non-deterministic
+        let nonDeterministicResultMatchesOneValue13 = iOS13Result == "value" || iOS13Result == "anotherValue"
+        let nonDeterministicResultMatchesOneValue12 = iOS12Result == "value" || iOS12Result == "anotherValue"
+        XCTAssertTrue(nonDeterministicResultMatchesOneValue13)
+        XCTAssertTrue(nonDeterministicResultMatchesOneValue12)
+    }
+    
+    func testResponseHttpHeaderNoMatch() throws {
+        // setup
+        let searchKey = "something-else"
+        let response = HTTPURLResponse(url: URL(string: "https://someurl")!, statusCode: 200, httpVersion: nil, headerFields: ["Some-Key": "value"])
+        let connection = HttpConnection(data: nil, response: response, error: nil)
+        
+        // test
+        let iOS13Result = connection.responseHttpHeader(forKey: searchKey)
+        let iOS12Result = connection.response?.lowercasedHeaders[searchKey.lowercased()] as? String
+        
+        // verify
+        XCTAssertNil(iOS13Result)
+        XCTAssertEqual(iOS13Result, iOS12Result)
+    }
+    
+    func testResponseHttpHeaderSpecialChars() throws {
+        // setup
+        let searchKey = "$om3-Ke?"
+        let response = HTTPURLResponse(url: URL(string: "https://someurl")!, statusCode: 200, httpVersion: nil, headerFields: ["$om3-Ke?": "value"])
+        let connection = HttpConnection(data: nil, response: response, error: nil)
+        
+        // test
+        let iOS13Result = connection.responseHttpHeader(forKey: searchKey)
+        let iOS12Result = connection.response?.lowercasedHeaders[searchKey.lowercased()] as? String
+        
+        // verify
+        XCTAssertEqual("value", iOS13Result)
+        XCTAssertEqual(iOS13Result, iOS12Result)
+    }
+    
+    func testResponseHttpHeaderMixedCase() throws {
```

Comment:
nit: you could reduce 3 tests into 1 by feeding in a list of inputs for all mixed case & mismatched cases and verifying the results